python_modules/libraries/dagster-dg/dagster_dg/docs.py

```diff
@@ -0,0 +1,168 @@
+import tempfile
+import webbrowser
+from dataclasses import dataclass
+from typing import AbstractSet, Any, Mapping, Optional, Sequence, Union
+
+import markdown
+import yaml
+
+from dagster_dg.context import RemoteComponentType
+
+REF_BASE = "#/$defs/"
+COMMENTED_MAPPING_TAG = "!commented_mapping"
+
+
+@dataclass(frozen=True)
+class CommentedObject:
+    key: str
+    value: Union[Sequence["CommentedObject"], Any]
+    comment: Optional[str]
+    top_level: bool = False
+
+
+def _dereference_schema(
+    json_schema: Mapping[str, Any], subschema: Mapping[str, Any]
+) -> Mapping[str, Any]:
+    if "$ref" in subschema:
+        return json_schema["$defs"].get(subschema["$ref"][len(REF_BASE) :])
+    else:
+        return subschema
+
+
+def _get_commented_obj(
+    name: str,
+    json_schema: Mapping[str, Any],
+    subschema: Mapping[str, Any],
+    available_scope: Optional[AbstractSet[str]] = None,
+) -> Union[CommentedObject, Any]:
+    additional_scope = subschema.get("dagster_available_scope")
+    available_scope = (available_scope or set()) | set(additional_scope or [])
+
+    subschema = _dereference_schema(json_schema, subschema)
+    if "anyOf" in subschema:
```

Comment:
do we need a todo here to handle the other `anyOf` cases? from what I can tell this just returns the first one

Reply:
added a TODO -- right now there aren't actually many true Union types, and the existing AnyOfs are just Optional fields, but of course that won't always be true